src/operators/throttleTime.ts

```diff
@@ -85,30 +85,38 @@ class ThrottleTimeSubscriber<T> extends Subscriber<T> {
   }
 
   protected _next(value: T) {
-    if (this.throttled) {
-      if (this.trailing) {
-        this._trailingValue = value;
-        this._hasTrailingValue = true;
-      }
-    } else {
-      this.add(this.throttled = this.scheduler.schedule(dispatchNext, this.duration, { subscriber: this }));
-      if (this.leading) {
-        this.destination.next(value);
-      }
+    if (!this.throttled && this.leading) {
+      this.throttle();
+      this.destination.next(value);
+    } else if (this.trailing) {
+      this._trailingValue = value;
+      this._hasTrailingValue = true;
+    }
+
+    if (!this.throttled) {
+      this.throttle();
     }
   }
 
+  private throttle(): void {
+    this.add(this.throttled = this.scheduler.schedule(dispatchNext, this.duration, { subscriber: this }));
+  }
+
   clearThrottle() {
     const throttled = this.throttled;
+
     if (throttled) {
+      throttled.unsubscribe();
+      this.remove(throttled);
+      this.throttled = null;
+
       if (this.trailing && this._hasTrailingValue) {
-        this.destination.next(this._trailingValue);
+        const trailingValue = this._trailingValue;
         this._trailingValue = null;
         this._hasTrailingValue = false;
+        this.throttle();
```

Comment:
this is unnecessary and changes the behavior of the operator. It's not supposed to start throttling again when the throttle is cleared.

Reply:
But this is exactly why I opened the PR. Because currently the operator emits 2 values each timeframe instead of 1.

If the doubleemit is intended, this PR can be closed and https://github.com/ReactiveX/rxjs/issues/2727 & https://github.com/ReactiveX/rxjs/issues/2859 rejected.

Reply:
hmm.. I think there's a disconnect here. `throttle()` actually starts the throttle measurement. The, with `leading` and `trailing` both `true`, it should emit the same as it would with `leading` behavior, but then emit the trailing value at the end.

Reply:
Oh, I okay, I actually see what you're saying now. This is to start the throttle again as soon as there's another emission (due to trailing behavior)

Reply:
Okay... the more I look at this, the more I think it will have to be a change that lands in v 6.0, because it will be a breaking change for someone. (Albeit a fix)